include/deal.II/matrix_free/matrix_free.h

```diff
@@ -519,6 +523,11 @@ class MatrixFree : public Subscriptor
      */
     bool cell_vectorization_categories_strict;
 
+    /**
+     * Assert that vectors passed to the MatrixFree loops are not ghosted.
```

Comment:
Maybe we can expand things a bit here? I'm thinking of something along the lines of:
```suggestion
     * Assert that vectors passed to the MatrixFree loops are not ghosted.
     * This variable is primarily intended to reveal bugs or performance
     * problems caused by vectors that are involuntarily in ghosted mode,
     * by adding a check that this is not the case. In terms of correctness,
     * the MatrixFree::loop() and MatrixFree::cell_loop() methods support
     * both cases and perform similar operations. In particular, ghost values are
     * always updated on the source vector within the loop, and the difference is
     * only in whether the initial non-ghosted state is restored.
```

Reply:
I have added the explanation!